contracts/multisig-prover/src/encoding/bcs.rs

```diff
@@ -0,0 +1,165 @@
+use bcs::to_bytes;
+use cosmwasm_std::HexBinary;
+use itertools::Itertools;
+
+use crate::error::ContractError;
+
+use super::Data;
+
+pub fn command_params(
+    source_chain: String,
+    source_address: String,
+    destination_address: String,
+    payload_hash: HexBinary,
+) -> Result<HexBinary, ContractError> {
+    let ret = to_bytes(&(
+        source_chain,
+        source_address,
+        <[u8; 32]>::try_from(&HexBinary::from_hex(&destination_address)?.to_vec()[..32])
+            .expect("couldn't convert destination_address to 32 byte array"), // TODO: is this right? Why are addresses 32 bytes?
+        payload_hash.to_vec(),
```

Comment:
Shouldn't it fail if `destination_address` has more than 30 bytes?

Reply:
https://doc.rust-lang.org/std/primitive.array.html#impl-TryFrom%3CVec%3CT,+Global%3E%3E-for-Box%3C%5BT;+N%5D,+Global%3E @cjcobb23 you don't need to slice it yourself.